peermanager/peermanager.go

```diff
@@ -83,8 +83,18 @@ func (pm *PeerManager) Disconnected(p peer.ID) {
 // GetProcess returns the process for the given peer
 func (pm *PeerManager) GetProcess(
 	p peer.ID) PeerProcess {
+	// Usually this this is just a read
+	pm.peerProcessesLk.RLock()
+	pqi, ok := pm.peerProcesses[p]
+	if ok {
+		pm.peerProcessesLk.RUnlock()
+		return pqi.process
+	}
+	pm.peerProcessesLk.RUnlock()
+	// but sometimes it involves a create (we still need to do get or create cause it's possible
+	// another writer grabbed the Lock first and made the process)
 	pm.peerProcessesLk.Lock()
-	pqi := pm.getOrCreate(p)
```

Comment:
It's possible that in between releasing the read lock, and taking the write lock, another process calls GetProcess().

So I think after taking the write lock we would need to check pqi again:
```
        pqi, ok := pm.peerProcesses[p]
	if ok {
            return pqi.process
        }
```

Reply:
check the code for getOrCeate -- it does this. (or the equivalent of it)

Reply:
Ah yes I see 👍